xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/pdfheader.vm

```diff
@@ -1,6 +1,8 @@
 #set ($title = "$!{pdfdoc.display('title', 'rendered', '', $pdfobj)}")
-#if ($title == '')
+#if ($title == '' && $doc.space != $tdoc.plainTitle)
   $!{escapetool.xml($doc.space)} - $!{escapetool.xml($tdoc.plainTitle)}
+#elseif ($title == '')
+  $!{escapetool.xml($tdoc.plainTitle)}
 #else
```

Comment:
Thanks for the PR!

I don't think I understand the new condition (maybe you could add some explanation?).

However the #if could be improved slightly with:

````
#if (title == '')
  #if (new condition)
    ...
  #else
   ...
#end
````

That should save an evaluation ($title == '') when the title is empty and the new condition is not satisfied.